core/src/services/analytics/mod.rs

```diff
@@ -0,0 +1,127 @@
+
+/// Module to support analysis on SBOMs
+pub mod sboms;
+
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use platform::persistence::s3;
+use std::fmt::Debug;
+use std::io::{BufReader, Cursor};
+use serde_json::Value;
+
+use crate::entities::packages::Package;
+use crate::{config, Error};
+
+fn get_file_name(provider_name: &str, purl: &str) -> String {
```

Comment:
So, the file name situation isn't great and that's because Snyk has to do it's weird thing.  I'm going to do my best to address that this weekend, but if I don't come up with a good plan, I will refactor this so that there is one, top level function that all of our code uses.

Reply:
I have this distilled to Trello, so I'm closing the conversation